File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.js
function solution(survey, choices) {
    console.log("지표", survey);
    console.log("선택지", choices);
    
    const data = {
        R: 0,
        T: 0,
        C: 0,
        F: 0,
        J: 0,
        M: 0,
        A: 0,
        N: 0,
    }
    
    const points = [ 3, 2, 1, 0, 1, 2, 3 ];
    
    if(survey.length !== choices.length){
        return "RCJA";
    }
    
    for(let i = 0; i < survey.length; i++){
        const [first, second] = survey[i].split('');
        
        const pick = choices[i] - 1;
        
        if(0 <= pick && pick <= 2){
            data[first] += points[pick];
        }
        if(4 <= pick && pick <= 6){
            data[second] += points[pick];
        }
    }
    
    let test = "";
    
    if(data['R'] >= data['T']){
        test += "R"
    } else {
        test += 'T'
    }
    if(data['C'] >= data['F']){
        test += "C"
    } else {
        test += 'F'
    }
    if(data['J'] >= data['M']){
        test += "J"
    } else {
        test += 'M'
    }
    if(data['A'] >= data['N']){
        test += "A"
    } else {
        test += 'N'
    }
    
    
    return test;
}

File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.py
def solution(survey, choices):
    metric_dict = {
        'RT': 0,
        'CF': 0, 
        'JM': 0,
        'AN': 0
    }
    
    point_dict = {
        1: 3, 
        2: 2,
        3: 1,
        4: 0,
        5: -1,
        6: -2,
        7: -3
    }
    
    for s, c in zip(survey, choices):
        [l, r] = s
        
        dir = 1 if metric_dict.get(s) is not None else -1
        
        metric = s if dir == 1 else r + l
        
        metric_dict[metric] += (point_dict[c] * dir)
    
    result = ""
    
    for m, p in metric_dict.items():
        [l, r] = m
        
        if p >= 0:
            result += l
        else:
            result += r
            
    return result